eng/pipelines/common/global-build-job.yml

```diff
@@ -22,7 +24,25 @@ jobs:
     workspace:
       clean: all
 
-    variables: ${{ parameters.variables }}
+    variables:
+    - name: osGroup
```

Comment:
Do these variables have to be named like that? I've seen problems where we somehow matched the MSBuild property name and since AzDo sets these variables as env vars, then you hit weird errors. Maybe append them with `_` or something to avoid clashes in between MSBuild and yml variables.

Reply:
Also it is kind of weird to define these variables here just to make extraStepTemplates of runtime lab work. 

I believe osGroup, osSubgroup, archType and buildConfig should be taken as parameters by extra step template. 

buildConfigUpper is already used in a couple of places and defined in `mono/xplat-job` and `coreclr/xplat-job`. 
https://github.com/dotnet/runtime/search?l=YAML&q=buildConfigUpper

I think we should instead remove it from there and move the variable declaration to `xplat-setup.yml` which is common to all jobs, that way we ensure it is always defined, no way to duplicate logic across multiple ymls.